.github/workflows/publish.yaml

```
@@ -65,3 +78,29 @@ jobs:
             labels: ${{ steps.meta.outputs.labels }}
             cache-from: type=gha
             cache-to: type=gha,mode=max
+
+        # Sign the resulting Docker image digest except on PRs.
+        # This will only write to the public Rekor transparency log when the Docker
+        # repository is public to avoid leaking data.  If you would like to publish
+        # transparency data even for private images, pass --force to cosign below.
+        # https://github.com/sigstore/cosign
+        - name: Sign the published Docker image
+          # This step uses the identity token to provision an ephemeral certificate
+          # against the sigstore community Fulcio instance.
+          run: echo "${{ steps.meta.outputs.tags }}" | xargs -I {} cosign sign --yes {}@${{ steps.build-and-push.outputs.digest }}
+    provenance-for-images:
+      needs: [paranoia-inception]
+      permissions:
+        actions: read # for detecting the Github Actions environment.
+        id-token: write # for creating OIDC tokens for signing.
+        packages: write # for uploading attestations.
+      uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.5.0
+      with:
+        image: ghcr.io/${{ github.repository }}
+        # The image digest is used to prevent TOCTOU issues.
+        # This is an output of the docker/build-push-action
+        # See: https://github.com/slsa-framework/slsa-verifier#toctou-attacks
+        digest: ${{ needs.paranoia-inception.outputs.digest }}
+        registry-username: ${{ github.actor }}
+      secrets:
+        registry-password: ${{ secrets.GITHUB_TOKEN }}
```

nit: newline at end of file
```suggestion
        registry-password: ${{ secrets.GITHUB_TOKEN }}

```